wfe2/wfe.go

```diff
@@ -2371,10 +2301,11 @@ func (wfe *WebFrontEndImpl) NewOrder(
 	// are authoritative. This saves us from adding latency to each request.
 	// Goroutines spun out below will respect a context deadline set by the
 	// ratelimits package and cannot be prematurely canceled by the requester.
-	var txns []ratelimits.Transaction
+	refundLimits := func() {}
 	if !limitsExempt {
-		txns = wfe.newNewOrderLimitTransactions(acct.ID, names)
-		go wfe.checkNewOrderLimits(ctx, txns)
+		go func() {
+			refundLimits = wfe.checkNewOrderLimits(ctx, acct.ID, names)
```

Comment:
These lines, combined with 2321/2325 below are a data race. The assignment to `refundLimits` can be executing concurrently with the read of `refundLimits` on 2321 / 2325.

I know we're putting this in a goroutine to avoid incurring latency cost before we actually turn the feature on. But since it's looking like we're incurring complexity by trying to do things this way, maybe it's time to drop the goroutine and call this in a blocking way? If we set a short timeout on the key-value rate limit checks, we can limit negative impacts in case of an outage.